mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,263 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import ast
+import json
+import os
+import re
+from botocore.vendored import requests
+import logging
+import secret_manager
+
+
+class LabelBot:
+    LABEL_PAGE_PARSE = 30  # Limit for total labels per page to parse
+
+    def __init__(self,
+                 repo=os.environ.get("repo"),
+                 github_user=None,
+                 github_oauth_token=None,
+                 apply_secret=True):
+        """
+        Initializes the Label Bot
+        :param repo: GitHub repository that is being referenced
+        :param github_user: GitHub username
+        :param github_oauth_token: GitHub authentication token (Personal access token)
+        :param apply_secret: GitHub secret credential (Secret credential that is unique to a GitHub developer)
+        """
+        self.repo = repo
+        self.github_user = github_user
+        self.github_oauth_token = github_oauth_token
+        if apply_secret:
+            self._get_secret()
+        self.auth = (self.github_user, self.github_oauth_token)
+        self.all_labels = None
```

Comment:
shall we provide a method? this variable is public but set through a protected method (in python pep8 terms)

Reply:
@larroy not sure what do you mean in here. This method is just meant to get all available labels on MXNet. It mainly used check if users provide a label that exists in MXNet and then applies them to issues and PRs.